test/unit/lookups/esri_test.rb

```diff
@@ -26,6 +28,23 @@ def test_query_for_geocode_with_token_for_storage
       res
   end
 
+  def test_query_for_geocode_with_client_credentials_for_storage
+    token_data = {
+      access_token: "xxxxx",
+      expires_in: 7200
+    }
+    stub_request(:post, "https://www.arcgis.com/sharing/rest/oauth2/token").to_return(:body => token_data.to_json)
```

Comment:
Instead of requiring a dependency for this test, stubbing one layer higher could also work. Please try stubbing this method: `Geocoder::EsriToken.generate_token(*configuration.api_key)`